oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/facets/ElasticSecureFacetAsyncProvider.java

```diff
@@ -112,16 +111,20 @@ public void endData() {
                                 .collect(Collectors.toList())
                         )
                 );
-        LOG.trace("End data {}", facetsMap);
+        LOG.trace("End data {}", facets);
```

Comment:
Maybe check if trace is enabled before calling LOG.trace() to avoid computing facets.toString(). `facets` is a map of lists, so the toString method may be expensive.

Reply:
there is no need for that. [Parameterized logging](https://www.tutorialspoint.com/slf4j/slf4j_parameterized_logging.htm) computes `toString` only when trace is enabled.